### PR TITLE
Filter / FilterButton improvements

### PR DESCRIPTION
**Pull request by egormesyats**
This would help so much for resolving this: 
[ANE-741](https://moovweb.atlassian.net/browse/ANE-741)
https://github.com/moovweb-projects/annieselke/pull/532

### Changes (non chronological order):
- https://github.com/moovweb/react-storefront/commit/68f52dd8fba40a4035d8be94695bb309c10d628e added `LoadMask` into `Filter`'s `facetGroups` block when model is loading
- https://github.com/moovweb/react-storefront/commit/ab80bd29a0dfe7c9674f6e7a7a3127f2e6348cb3 disable `clear all` button when model is loading
- https://github.com/moovweb/react-storefront/commit/92fcce172d837caf214b8270f01ba865023d7531 `clear all` button semantics fix: use `<button>` instead `<a>` w/o `href` attribute 

### UPD:
- https://github.com/moovweb/react-storefront/pull/111/commits/29ffc546da3b05b50ae7cd007139590d49fcaf9f tech debt: dynamically change classes via `classnames`

### UPD 2:
- https://github.com/moovweb/react-storefront/pull/111/commits/0f03f69069e6c3b5862db04262ab1f667a8827cd Revert for previous 5.x release issue. 
I found a workaround to fix [ANE-724](https://moovweb.atlassian.net/browse/ANE-724) issue from our side: 
https://github.com/moovweb-projects/annieselke/pull/532/commits/b49b35044902e41b006f43b1633f8dc25b837ad4

### UPD 3. Tests added:
- https://github.com/moovweb/react-storefront/pull/111/commits/1f754d01edf08e660c03e281e589cf49b820a38f Filter.test snapshots fix
- https://github.com/moovweb/react-storefront/pull/111/commits/0ff6b74b8d2649eec69bd460671041b192b64bb9 FilterButton.test snapshots fix
- https://github.com/moovweb/react-storefront/pull/111/commits/ea1132b6b49e915612ae8e6ba1bc84df183d2271 FilterButton.test: fix some tests

_Note:_
_To simplify review process you can review by single commits_

### CODE REVIEW

File: packages/react-storefront/src/FilterButton.js [hunk 1]
```diff
@@ -111,7 +117,12 @@ export default class FilterButton extends Component {
               <Hbox justifyContent="center">
                 <div>{title}</div>
                 { hideClearLink || model.filters.length === 0 ? null : (
-                  <a className={clear} onClick={model.clearAllFilters}>{clearLinkText}</a>
+                  <button 
+                    className={`${clear} ${model.loading ? clearDisabled : ''}`}
```
Comment (kevhender) [inline packages/react-storefront/src/FilterButton.js]: Use `classnames` here.

Comment (egormesyats) [reply packages/react-storefront/src/FilterButton.js]: @kevhender seems this dependency it's not included in RSF package

Comment (kevhender) [reply packages/react-storefront/src/FilterButton.js]: It's used all over the place though, so it's in there as a dependency already. You should probably add it to the package.json, though.

Comment (markbrocato) [reply packages/react-storefront/src/FilterButton.js]: Yeah, you can just add that to the package.json in react-storefront/packages/react-storefront and run a yarn install on the root of the monorepo.  It should have been there already.  We use it throughout RSF.

Comment (egormesyats) [reply packages/react-storefront/src/FilterButton.js]: OK @kevhender @markbrocato got it guys

**Comment by markbrocato** [comment]
@egormesyats There is a bug with the layout in the filter drawer.  See my message in Slack.

**Pull request merged**